### PR TITLE
feat(pacman static): static linking pacman

### PR DESCRIPTION
**Pull request by lseman**
Static linking pacman with a v3 compiled sqlite3 inside default CachyOS docker.

### CODE REVIEW

**Comment by vnepogodin** [comment]
We shouldnt provide binaries.
And if then we need binaries for v1 and v3.
Also staging shouldnt be enabled

File: pacman/PKGBUILD [hunk 1]
```diff
@@ -47,23 +46,28 @@ source=(
...
 fi
 
 prepare() {
+  cp libsqlite3.a "${pkgname}"
   cd "${pkgname}"
   # we backport way too often in pacman
   # lets at least make it more convenient
   local src
   for src in "${source[@]}"; do
-    src="${src%%::*}"
```
Comment (vnepogodin) [inline pacman/PKGBUILD:62]: Spacing got wrong. Also looks like you are using tabs instead of spaces here

**Comment by lseman** [comment]
The easiest way to provide non-binary sqlite static would be to provide a custom sqlite in CachyOS repos configure as:

 100   │   make clean
 101   │   ./configure --prefix=/usr \
 102   │   --enable-shared \
 103   │   --enable-static \
 104   │     --enable-fts3 \
 105   │     --enable-fts4 \
 106   │     --enable-fts5 \
 107   │     --enable-rtree \
 108   │     TCLLIBDIR=/usr/lib/sqlite$pkgver \

which will also provide the .a, and then using this to link.

**Comment by vnepogodin** [comment]
> The easiest way to provide non-binary sqlite static would be to provide a custom sqlite in CachyOS repos configure as:
> 
>  100   │   make clean
>  101   │   ./configure --prefix=/usr \
>  102   │   --enable-shared \
>  103   │   --enable-static \
>  104   │     --enable-fts3 \
>  105   │     --enable-fts4 \
>  106   │     --enable-fts5 \
>  107   │     --enable-rtree \
>  108   │     TCLLIBDIR=/usr/lib/sqlite$pkgver \
> 
> which will also provide the .a, and then using this to link.

Nah. This should be inside the pacman pkgbuild

**Comment by lseman** [comment]
Make pacman PKGBUILD build a sqlite?

**Comment by vnepogodin** [comment]
> Make pacman PKGBUILD build a sqlite?

Yes when staging is enabled

**Comment by lseman** [comment]
Adding sqlite to PKGBUILD if staging is enabled, getting latest sqlite version from Archlinux API.

File: pacman/PKGBUILD [hunk 2]
```diff
@@ -79,8 +79,14 @@ if $_staging; then
...
-  vSQLITE3=$(echo $vSQLITE3 | awk -F. -v len=$target_length '{printf "%d%02d%02d", $1, $2, $3; for(i=length($0)+1; i<=len; i++) printf "0"}')
+  # from sqlite website:
+  # The version is encoded so that filenames sort in order of increasing version number when viewed using "ls".
+  # For version 3.X.Y the filename encoding is 3XXYY00. For branch version 3.X.Y.Z, the encoding is 3XXYYZZ. 
+  vSQLITE3=$(echo $vSQLITE3 | awk -F. '{ 
+      if (NF == 3) { printf "%d%02d%02d00", $1, $2, $3 }
+      else if (NF == 4) { printf "%d%02d%02d%02d", $1, $2, $3, $4 }
+  }')
+  echo "Using sqlite3 version $vSQLITE3"
   wget https://www.sqlite.org/2023/sqlite-src-${vSQLITE3}.zip
```
Comment (vnepogodin) [inline pacman/PKGBUILD:90]: Should be in source array

**Review by vnepogodin — CHANGES_REQUESTED** [review]

**Pull request merged**